taworks/taform/views.py

```diff
@@ -214,8 +214,17 @@ def apply(request):
                 }
 
         try:
+            studentID=str(request.POST['student_id'])
+            apps_made= models.Application.objects.filter(student__student_id=studentID, preference__in = [1,2,3]).count()
+            app_id = 0
+            if apps_made > 0:
+                previous_submissions = True
+            else:
+                previous_submissions = False
+
             citizenship=str(request.POST['citizenship'])
-            if citizenship == 'Student Visa':
+            visa_expiration=str(request.POST['student_visa_expiry_date'])
+            if (citizenship == 'Student Visa' and not visa_expiration):
```

Comment:
nice =)
you can also check visa_expiration is in the format 
yyyy-mm-dd
but you don't have to since s_form.is_valid() takes care of that